matchzoo/models/esim.py

```diff
@@ -0,0 +1,218 @@
+        )
+        params['optimizer'] = 'adam'
+        params.add(Param(name='num_layers', value=1,
+                         desc="Number of LSTM layers."))
+        params.add(Param(
+            'dropout_rate', 0.0,
+            hyper_space=hyper_spaces.quniform(
+                low=0.0, high=0.8, q=0.01),
+            desc="The dropout rate."
+        ))
+        params.add(Param(
+            name='lstm_units',
+            value=256,
+            desc="The hidden dimension in LSTM cell."
+        ))
+        return params
+
+    def build(self):
+        """
+        Build model structure.
+
+        """
+        # Scalar dimensions referenced here:
+        #   B = batch size (number of sequences)
+        #   D = embedding size
+        #   L = `input_left` sequence length
+        #   R = `input_right` sequence length
+        #   H = LSTM hidden size
+
+        # Left input and right input
+        # shape = [B, L]
+        # shape = [B, R]
+        input_left, input_right = self._make_inputs()
+        hidden_size = self._params['lstm_units']
+
+        def tile_tensor(x):
+            res = K.tile(K.expand_dims(x), [1, 1, 2 * hidden_size])
+            return res
+
+        embedding = self._make_embedding_layer()
+        # Look up embedding matrix and get embed representation
+        # shape = [B, L, D]
+        embed_left = embedding(input_left)
+        # shape = [B, R, D]
+        embed_right = embedding(input_right)
+        # Get the mask for calculating attention
+        # shape = [B, L]
+        left_mask = keras.layers.Lambda(get_mask)(input_left)
+        # left_mask = keras.layers.Lambda(lambda x: K.tf.Print(x, [x]))(left_mask)
+
+        # shape = [B, R]
+        right_mask = keras.layers.Lambda(get_mask)(input_right)
+
+        # Add dropout layer
+        embed_left = keras.layers.Dropout(
+            rate=self._params['dropout_rate'])(embed_left)
+        embed_right = keras.layers.Dropout(
+            rate=self._params['dropout_rate'])(embed_right)
+
+        encoder_layer = keras.layers.Bidirectional(keras.layers.LSTM(
+            hidden_size,
+            return_sequences=True,
+            dropout=self._params['dropout_rate']
+        ))
+        # Encode the two text
+        # shape = [B, L, 2*H]
+        encoded_left = encoder_layer(embed_left)
+        # shape = [B, R, 2*H]
+        encoded_right = encoder_layer(embed_right)
+        # Make interaction and get the interactive representation
+        # shape = [B, L, 2*H]
+        # shape = [B, R, 2*H]
+        attended_left, attended_right = self._attention(encoded_left, left_mask, encoded_right, right_mask)
+        # shape = [B, L, 8*H]
+        enhanced_left = keras.layers.Concatenate()([encoded_left, attended_left,
+                                                    keras.layers.Subtract()([encoded_left, attended_left]),
+                                                    keras.layers.Multiply()([encoded_left, attended_left])])

```

Comment:
I guess you do not need to  call `keras.layers.Substract` or `keras.layers,Multiply` or even `keras.layers.ReLU()` explicitly, use `substract`, `multiply` and `relu`  (string format) directly, which is more readable.